node-lib/src/config_files/p2p.rs

```diff
@@ -114,14 +114,17 @@ impl From<P2pConfigFile> for P2pConfig {
             sync_stalling_timeout,
             node_type,
             force_dns_query_if_no_global_addresses_known,
-        } = config_file;
+        } = self;
 
-        P2pConfig {
+        let boot_nodes = resolve_all(boot_nodes.unwrap_or_default().into_iter()).await?;
+        let reserved_nodes = resolve_all(reserved_nodes.unwrap_or_default().into_iter()).await?;
```

Comment:
I wonder if we should use `resolve_all` or `resolve_all_take_first` here; the first function returns all addresses for each name and the second one only returns one address for each name.

Reply:
It's wrong to do the name resolution at the config spot. The nodes should be stored as hostnames and resolved when the connection is to be made. This is the whole point of domain names + TTL. We're basically ignoring TTL by doing this.

Reply:
Yes, but I was assuming that it would require a large enough refactoring, which we wouldn't want to do. Or would we?

Reply:
I mean, we shouldn't do it incorrectly... that big refactoring? Isn't it just a change of a type?